src/Modules/UngradedModules/SubquoModule.jl

```diff
@@ -2190,3 +2199,65 @@ function is_known(::typeof(is_zero), M::SubquoModule)
   return false
 end
 
+#########################
+###Krull dimension
+#########################
+
+@doc raw"""
+    function krull_dim(M::ModuleFP)
+
+Return the Krull dimension of `M`.
+
+# Examples
+
+```jldoctest
+julia> R, (x, y, z) = polynomial_ring(QQ, [:x, :y, :z]);
+
+julia> A = R[x; y]
+[x]
+[y]
+
+julia> B = R[x^2; x*y; y^2; z^4]
+[x^2]
+[x*y]
+[y^2]
+[z^4]
+
+julia> M = SubquoModule(A, B)
+Subquotient of submodule with 2 generators
+  1: x*e[1]
+  2: y*e[1]
+by submodule with 4 generators
+  1: x^2*e[1]
+  2: x*y*e[1]
+  3: y^2*e[1]
+  4: z^4*e[1]
+
+julia> krull_dim(M)
+0
+
+```
+"""
+@attr Union{Int,NegInf} function krull_dim(M::ModuleFP)
+  ann = annihilator(M)
+  return krull_dim(ann)
+end
+
+@attr Int function krull_dim(M::SubquoModule{T}) where {T <: MPolyRingElem}
+  F = ambient_free_module(M)
+
+  if !all(repres(v) == F[i] for (i, v) in enumerate(gens(M)))
+    M = present_as_cokernel(M)
+  end
+
+  gb = groebner_basis(M.quo)
+  return Singular.dimension(singular_generators(gb))
+end
+
+@attr Union{Int,NegInf} function krull_dim(F::FreeMod)
+  if rank(F) == 0
+    return NegInf
```

Comment:
This is not correct, as it returns a type, not an instance of that type. Need to do either
```suggestion
    return NegInf()
```
or
```suggestion
    return -inf
```